pkg/ociinstaller/ocidownloader.go

```diff
@@ -27,27 +35,86 @@ func NewOciDownloader() *ociDownloader {
+	memoryStore := memory.New()
+	fileStore, err := file.NewWithFallbackStorage(destDir, memoryStore)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
 	defer fileStore.Close()
 
-	hybridStore := newHybridStore(fileStore)
-	pullOpts := []oras.PullOpt{
-		oras.WithAllowedMediaTypes(append(mediaTypes, MediaTypeConfig, MediaTypePluginConfig)),
-		oras.WithPullEmptyNameAllowed(),
+	// Connect to the remote repository
+	repo, err := remote.NewRepository(ref)
+	if err != nil {
+		return nil, nil, nil, nil, err
 	}
-	desc, layers, err := oras.Pull(ctx, o.resolver, ref, hybridStore, pullOpts...)
+
+	// Get credentials from the docker credentials store
+	storeOpts := credentials.StoreOptions{}
+	credStore, err := credentials.NewStoreFromDocker(storeOpts)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	// Prepare the auth client for the registry and credential store
+	repo.Client = &auth.Client{
+		Client:     retry.DefaultClient,
+		Cache:      auth.DefaultCache,
+		Credential: credentials.Credential(credStore), // Use the credential store
+	}
+
+	// Copy from the remote repository to the file store
+	log.Println("[TRACE] ociDownloader.Pull:", "pulling...")
+
+	copyOpt := oras.DefaultCopyOptions
+	// TODO: use WithTargetPlatform to limit downloads
+	// copyOpt.WithTargetPlatform(&ocispec.Platform{
+	// 	Architecture: "amd64",
+	// 	OS:           "linux",
+	// })
+	manifestDescriptor, err := oras.Copy(ctx, repo, tag, fileStore, tag, copyOpt)
 	if err != nil {
-		return &desc, nil, nil, layers, err
+		log.Println("[ERROR] ociDownloader.Pull:", "failed to pull", ref, err)
+		return &manifestDescriptor, nil, nil, nil, err
 	}
-	configDesc, configData, err := hybridStore.GetConfig()
-	return &desc, &configDesc, configData, layers, err
+	log.Println("[TRACE] ociDownloader.Pull:", "manifest", manifestDescriptor.Digest, manifestDescriptor.MediaType)
+
+	// FIXME: this seems redundant as oras.Copy() already downloads all artifacts, but that's the only I found
+	// to access the manifest config. Also, it shouldn't be an issue as files are not re-downloaded.
+	manifestJson, err := content.FetchAll(ctx, fileStore, manifestDescriptor)
+	if err != nil {
+		log.Println("[TRACE] ociDownloader.Pull:", "failed to fetch manifest", manifestDescriptor)
+		return &manifestDescriptor, nil, nil, nil, err
+	}
+	log.Println("[TRACE] ociDownloader.Pull:", "manifest content", string(manifestJson))
+
+	// Parse the fetched manifest
+	var manifest ocispec.Manifest
+	err = json.Unmarshal(manifestJson, &manifest)
+	if err != nil {
+		log.Println("[TRACE] ociDownloader.Pull:", "failed to unmarshall manifest", manifestJson)
+		return &manifestDescriptor, nil, nil, nil, err
+	}
+
+	// Fetch the config from the file store
+	configData, err := content.FetchAll(ctx, fileStore, manifest.Config)
+	if err != nil {
+		log.Println("[ERROR] ociDownloader.Pull:", "failed to fetch config", manifest.Config.MediaType, err)
+		return &manifestDescriptor, nil, nil, nil, err
```

Comment:
We should probably return `nil` instead of `&manifestDescriptor` in case of errors, as it is not used from the calling side:
```suggestion
		return nil, nil, nil, nil, err
```

Reply:
Done.